spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -842,7 +842,7 @@ struct ProductionVisitor : public production::Visitor {
             ncur_max_size = state().cur;
         }
 
-        else if ( auto a = field->attributes()->find("&size") )
+        else if ( auto a = field->attributes()->find("&size"); a && ! field->attributes()->find("&eod") )
```

Comment:
I'm unclear why this (and below) is needed for the problem being fixed here. Mind extending the commit message a bit to explain?

Reply:
Added the following to the commit message:

> We also tweak the implementation of the check for consumed `&size` when used together with `&eod`: if the `&size` was already nested in a limited view a larger `&size` value could previously extend the view so the `&eod` effectively was ignored. Since we now do not extend the `View` anymore we need to only activate the check for consumed `&size` if `&eod` was not specified since in this case the user communicated that they are fine with consuming less data.

The reason this worked correctly if a field with `&size` & `eod` was not nested is due to how we update `ncur`.